CMakeLists.txt

```diff
@@ -0,0 +1,116 @@
+cmake_minimum_required(VERSION 3.11)
+
+set(PRODUCT_NAME cretro)
+string(TOUPPER ${PRODUCT_NAME} MSG_PREFIX)
+
+project(
+    ${PRODUCT_NAME}
+    LANGUAGES C
+)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_C_COMPILER gcc)
+
+set(INSTALL_PATH ~/.local/bin/${PRODUCT_NAME})
+
+add_compile_options(
+    -Wall
+    -Wextra
+    -Werror
+    -Wunused
+    -Wshadow
+    -Wcast-align
+    -Wconversion
+    -Wlogical-op
+    -Wdouble-promotion
+    -fstack-protector
+    -Wstack-protector
+    -Wredundant-decls
+    -Wpacked
+    -Wmissing-declarations
+    -Wmissing-prototypes
+    -Wstrict-prototypes
+    -Wno-aggressive-loop-optimizations
+    -Wjump-misses-init
+    -Wwrite-strings
+    -Wcast-qual
+    -Wundef
+    -Wunsafe-loop-optimizations
+    -Wfloat-equal
+    -Wtrampolines
+    -Warray-bounds=2
+    -Wsuggest-attribute=cold
+    -Wsuggest-attribute=malloc
+    -Wsuggest-attribute=pure
+    -Wsuggest-attribute=const
+    -Wsuggest-attribute=noreturn
+    -Wsuggest-attribute=format
+    -Wstrict-overflow=4
+    -Wswitch-enum
+    -Wswitch-default
+    -Wmissing-include-dirs
+    -Wformat=2
+)
+
+if (CMAKE_BUILD_TYPE STREQUAL "Debug" OR CMAKE_BUILD_TYPE STREQUAL "DEBUG" OR CMAKE_BUILD_TYPE STREQUAL "debug")
+    message("[${MSG_PREFIX}] Using -g")
+    add_compile_options(-g)
+else()
+    message("[${MSG_PREFIX}] Using -O3")
+    add_compile_options(-O3)
+endif()
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
+    message("[${MSG_PREFIX}] Using clang")
+    add_compile_options(
+        -Weverything
+        -Wno-unknown-warning-option
+        -Wno-format-nonliteral
+        -ferror-limit=0
+    )
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_C_COMPILER_ID STREQUAL "GNU")
+    message("[${MSG_PREFIX}] Using gcc")
+    add_compile_options(
+        ### WARNINGS
+        # Standard GCC warnings: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html
+        -Werror
+        -Wlogical-op
+        -Wjump-misses-init
+        -Wunsafe-loop-optimizations
+        -Wtrampolines
+        -Wsuggest-attribute=cold
+        -Wsuggest-attribute=malloc
+        -Wsuggest-attribute=pure
+        -Wsuggest-attribute=const
+        -Wsuggest-attribute=noreturn
+        -Wsuggest-attribute=format
+        -Wformat=1
+        -fmax-errors=0
+    )
+endif()
+
+add_link_options(
+    -Wall
+    -lm
+    -lSDL2
+)
+
+include_directories(src)
+set(SOURCE_FILES
```

Comment:
I would suggest to use some pattern matching to collect all files. It's really cumbersome to add them manually.

Reply:
Adding the files manually is the suggested way to add source files to a target (see https://cmake.org/cmake/help/latest/command/file.html?highlight=glob#glob)

Reply:
I see 🤔 Apparently that's necessary to correctly track changes. In combination with Ninja that should indeed be better than using GLOB